[Bug]: 23.5.3 causes build errors

### Minimal, reproducible example

```TypeScript
Simply updating from Puppeteer 23.5.2 to 23.5.3 causes a build error (s.b.).
```


### Background

Error comes up when running Angular (18.2.8) E2E tests (TS 5.5.3).

### Expectation

No build issues.

### Reality

>Building tests 🛠️ ... 
>
>node_modules/puppeteer/lib/types.d.ts:1:35 - error TS2307: Cannot find module 'child_process' or its corresponding type declarations.
>
>1 import type { ChildProcess } from 'child_process';
                                    ~~~~~~~~~~~~~~~
>
>node_modules/puppeteer/lib/types.d.ts:3:29 - error TS2307: Cannot find module 'stream' or its corresponding type declarations.
>
>3 import { PassThrough } from 'stream';
                              ~~~~~~~~
>
>node_modules/puppeteer/lib/types.d.ts:3527:23 - error TS2591: Cannot find name 'Buffer'. Do you need to install type definitions for node? Try `npm i --save-dev @types/node` and then add 'node' to the types field in your tsconfig.
>
>3527     buffer(): Promise<Buffer>;
                           ~~~~~~
>
>
>Found 3 errors in the same file, starting at: node_modules/puppeteer/lib/types.d.ts:1

### Puppeteer configuration file (if used)

_No response_

### Puppeteer version

23.5.3

### Node version

20.18.0

### Package manager

npm

### Package manager version

10.8.2

### Operating system

Linux

--- a/.eslintrc.js
+++ b/.eslintrc.js
@@ -57,7 +57,7 @@ module.exports = {
       'error',
       'always',
       {
-        markers: ['*'],
+        markers: ['*', '/'],
       },
     ],
     eqeqeq: ['error'],

--- a/packages/puppeteer-core/api-extractor.json
+++ b/packages/puppeteer-core/api-extractor.json
@@ -14,6 +14,7 @@
   "dtsRollup": {
     "enabled": true,
     "untrimmedFilePath": "",
+    "omitTrimmingComments": true,
     "alphaTrimmedFilePath": "lib/types.d.ts"
   },
 

--- a/packages/puppeteer-core/src/api/Browser.ts
+++ b/packages/puppeteer-core/src/api/Browser.ts
@@ -3,7 +3,7 @@
  * Copyright 2017 Google Inc.
  * SPDX-License-Identifier: Apache-2.0
  */
-
+/// <reference types="node"  preserve="true"/>
 import type {ChildProcess} from 'child_process';
 
 import type {Protocol} from 'devtools-protocol';

